### PR TITLE
Add per-disk seclabel support

### PR DESCRIPTION
**Pull request by waynesun09**
As libvirt xml support per-disk seclabel setting, implement seclabel class which include multiple element texts and attributes.
Handle the element texts in marshal_from_seclabel and XMLElementList, update both to handle extra dict under key 'text'.
e.g.
There are text element 'label' and 'baselabel' in seclabel:
```
      <source path='/tmp/file'>
        <seclabel model='dac' relabel='no'>
          <label>0:0</label>
          <baselabel>0:0</baselabel>
        </seclabel>
      </source>
```
both text element under parent tag 'seclabel' and with text value.
the return tuple of marshal_from_seclabel will be:
```
('seclabel', {'text': {'label': '0:0', 'baselabel': '0:0'}, 'model': 'dac', 'relabel': 'yes'})
```
and XMLElementList will create child element text after create seclabel element with attribute.

This update is different to PR:
https://github.com/autotest/virt-test/pull/1902

which handle only one text element just under the parent tag. 

e.g.
the text element hostname with only one value, and multiple hostname elements.
```
          <host ip='192.168.122.2'>
            <hostname>myhost</hostname>
            <hostname>myhostalias</hostname>
          </host>
```

the 2 return tuple of marshal_from_hostname will be
```
("hostname", {}, 'myhost')
```
and 
```
("hostname", {}, 'myhostalias')
```
then handle by XMLElementList to create element without attribute but with text under tag hostname.

### CODE REVIEW

**Comment by yumingfei** [comment]
Hey, @waynesun09 , could you update it against @rbian's PR?

**Comment by waynesun09** [comment]
@yumingfei yeah, just found out 1902 got merged. I'll rebase this, thx.

**Comment by waynesun09** [comment]
@yumingfei rebase done

File: virttest/libvirt_xml/accessors.py [hunk 1]
```diff
@@ -802,15 +802,35 @@ def __call__(self, value):
...
+                    new_dict,
+                    text)
+
+                # To support child element contains text, create sub element
+                # with text under new created parent element.
+                if 'text' in element_tuple[1].keys():
+                    text_dict = element_tuple[1]['text']
+                    attr_dict = {}
+                    for text_key, text_val in text_dict.items():
+                        child_element = xml_utils.ElementTree.SubElement(
```
Comment (yumingfei) [inline virttest/libvirt_xml/accessors.py]: It seems that child_element parameter is useless.

Comment (yumingfei) [inline virttest/libvirt_xml/accessors.py]: It seems that child_element is useless.

Comment (waynesun09) [inline virttest/libvirt_xml/accessors.py]: @yumingfei yes, updating.

**Comment by yumingfei** [comment]
Acked.

**Comment by rbian** [comment]
Why not use XMLElementNest to implement a subelement?

**Comment by waynesun09** [comment]
@rbian 'seclabel' under disk could be multiple and XMLElementList could handle multiple child-elements with same tag.

e.g.
```
      <source path='/tmp/file'>
        <seclabel model='dac' relabel='yes'>
          <label>0:0</label>
        </seclabel>
        <seclabel model='selinux' relabel='yes'>
          <label>system_u:object_r:virt_image_t:s0</label>
        </seclabel>
      </source>
```

**Comment by rbian** [comment]
Okay, understand now.

**Comment by yumingfei** [comment]
OK, applied, Thanks @rbian, @waynesun09

**Pull request merged**